huaweicloud/services/sdrs/resource_huaweicloud_sdrs_protection_group.go

```diff
@@ -0,0 +1,254 @@
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	n, err := protectiongroups.Get(client, d.Id()).Extract()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.1013" {
+				// `SDRS.1013` means protection group not found
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error retrieving SDRS protection group")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("name", n.Name),
+		d.Set("description", n.Description),
+		d.Set("source_availability_zone", n.SourceAZ),
+		d.Set("target_availability_zone", n.TargetAZ),
+		d.Set("domain_id", n.DomainID),
+		d.Set("source_vpc_id", n.SourceVpcID),
+		d.Set("dr_type", n.DrType),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceProtectionGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	if d.HasChange("name") {
+		updateOpts := protectiongroups.UpdateOpts{
+			Name: d.Get("name").(string),
+		}
+		if _, err = protectiongroups.Update(client, d.Id(), updateOpts).Extract(); err != nil {
+			return diag.Errorf("error updating SDRS protection group name, %s", err)
+		}
+	}
+
+	if d.HasChange("enable") {
+		var enableErr error
+		if d.Get("enable").(bool) {
+			enableErr = enableAndWaitForProtectionGroupSuccess(d, client)
+		} else {
+			enableErr = disableAndWaitForProtectionGroupSuccess(d, client)
+		}
+		if enableErr != nil {
+			return diag.FromErr(enableErr)
+		}
+	}
+	return resourceProtectionGroupRead(ctx, d, meta)
+}
+
+func resourceProtectionGroupDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	n, err := protectiongroups.Delete(client, d.Id()).ExtractJobResponse()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.0207" {
+				// `SDRS.0207` means invalid protection group ID
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{},
+					"error deleting SDRS protection group")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	deleteTimeoutSec := int(d.Timeout(schema.TimeoutDelete).Seconds())
+	if err = protectiongroups.WaitForJobSuccess(client, deleteTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```

Comment:
```suggestion
	d.SetId("")
	return nil
```

Reply:
The deletion method will automatically set the ID to “”.